vpr/src/route/rr_graph_fpga_interchange.cpp

```diff
@@ -376,13 +376,33 @@ struct InterchangeRRGraphBuilder {
     /** @brief Creates mapping from tile_id to its location
      */
     void create_tile_id_to_loc() {
-        int max_height = 0;
         for (const auto& tile : ar_.getTileList()) {
             tile_loc_bimap_.insert(tile.getName(), t_location(tile.getCol() + 1, tile.getRow() + 1));
-            max_height = std::max(max_height, (int)(tile.getRow() + 1));
         }
+
+        // Find the constant tile via the utility lambda
+        auto find_constant_tile = [&]() {
+            const auto& grid = device_ctx_.grid;
+            for (size_t y=0; y<grid.height(); ++y) {
+                for (size_t x=0; x<grid.width(); ++x) {
+                    const auto* type = grid[x][y].type;
+                    if (type != nullptr) {
+                        if (!strcmp(type->name, str(-1).c_str())) {
```

Comment:
A side improvement that could be added to this PR would be a #define/const/constexpr for constant block index. I had to look up the implementation of `str` in order to understand this line.